node/mock-test.ts

```diff
@@ -125,4 +123,195 @@ export class MockTestRunner {
             console.log('TRACE FILE: ' + traceFile);
         }
     }
-}
+
+    // Returns a path to node.exe with the correct version for this task (based on if its node10 or node)
+    private getNodePath(): string {
+        const version: number = this.getNodeVersion();
+
+        // Check if version needed can be found on the path.
+        let nodePath: string = shelljs.which('node');
```

Comment:
let's take out this fuzzy check on current node major version.  we want a crisp run of exactly the versions used by the agent.  this isn't optimizing anything since we're caching in the home dir per above.

Reply:
I left an exact match check because it saves a download on the first run (and thus all CI runs) if its already stored in the path. I'd be comfortable taking it out if you feel strongly though.

Reply:
Taken out per offline discussion